src/sentry/templates/sentry/partial/preload-data.html

```diff
@@ -1,4 +1,8 @@
 {% load sentry_assets %}
+{% load sentry_trace %}
+
+<meta name="sentry-trace" content="{% get_sentry_trace %}">
+<meta name="baggage" content="{% get_sentry_baggage %}">
```

Comment:
Do you need the meta tags if you're also setting the headers on the xhr requests?

Reply:
@AbhiPrasad can you confirm this?

Reply:
@markstory Yes. We need the meta tags since the the JS SDK will use this information to propagate the trace id.

It's propagated here: https://github.com/getsentry/sentry-javascript/blob/a063fbccd79cab77da627e45166e50ddcd96928c/packages/tracing-internal/src/browser/browsertracing.ts#L304-L305

Otherwise we will have 2 disconnected trace ids.

Reply:
> It's propagated here: 

Neat! TIL. Thanks for the extra context.